src/main/java/parser/Parser.java

```diff
@@ -317,63 +321,71 @@ private String[] getSalesArguments(String userInput) {
+     * @throws InvalidViewCommandException if an invalid option flag is provided
+     */
     private Command processViewCommand(String userInput) throws TantouException {
         userInput = removeViewPrefix(userInput);
         String authorName = null;
-        boolean hasDeadlineFlag = false;
+        boolean hasByDateFlag = false;
         boolean hasSalesFlag = false;
 
-        Pattern authorPattern = Pattern.compile("-a\\s+((?:[^\\s-]+(?:\\s+|-))*[^\\s-]+)");
-        Matcher matcher = authorPattern.matcher(userInput);
+        Matcher authorNameFinder = AUTHOR_NAME_EXTRACT0R_PATTERN.matcher(userInput);
+        if (authorNameFinder.find()) {
+            authorName = authorNameFinder.group(0).trim();
+            if (authorName.isEmpty()) {
+                throw new NoAuthorProvidedException();
+            }
 
-        if (matcher.find()) {
-            authorName = matcher.group(1).trim();
-            userInput = userInput.replace(matcher.group(0), "").trim();  // Remove author part from userInput
+            userInput = userInput
+                    .replace(authorNameFinder.group(0), EMPTY_REGEX)    // Remove author name part from userInput
+                    .replace(AUTHOR_OPTION, EMPTY_REGEX)                // Remove AUTHOR_OPTION from userInput
+                    .trim();
         }
 
-        String[] tokens = userInput.trim().split("\\s+");
-
-        for (int i = 0; i < tokens.length; i++) {
-            String token = tokens[i].trim();
-
-            if (tokens[i].isBlank()) {
-                continue;
-            }
-
+        String[] tokens = userInput.split(ANY_SPACE_REGEX);  // Split the remaining around any number of spaces
+        for (String token : tokens) {
             switch (token) {
-            case "-d":
-                hasDeadlineFlag = true;
+            case BY_DATE_OPTION:
+                hasByDateFlag = true;
                 break;
-            case "-s":
+            case SALES_OPTION:
                 hasSalesFlag = true;
                 break;
-            case "-a":
-                // If control reaches here without getting caught by matcher, something invalid was entered
-                throw new NoAuthorProvidedException();
+            case EMPTY_REGEX:
+                break;
             default:
-                throw new TantouException("Unknown argument: " + tokens[i]);
+                throw new InvalidViewCommandException(token);
+
             }
         }
 
-        if ((hasDeadlineFlag || hasSalesFlag) && authorName == null) {
+        if ((hasByDateFlag || hasSalesFlag) && authorName == null) {
             throw new NoAuthorProvidedException();
         }
         if (authorName == null) {
             return new ViewAuthorsCommand();
         }
-        return new ViewMangasCommand(authorName, hasDeadlineFlag, hasSalesFlag);
+        return new ViewMangasCommand(authorName, hasByDateFlag, hasSalesFlag);
     }
```

Comment:
Overall, a really ingenious way of getting our arguments from the user. Will looking into SLAPping this.